osu.Game.Tests/Visual/Editing/TestSceneOpenEditorTimestamp.cs

```diff
@@ -0,0 +1,328 @@
+            AddAssert("recieved 'failed to process'", () =>
+                Game.Notifications.AllNotifications.Count(x => x.Text == EditorStrings.FailedToProcessTimestamp) == 2
+            );
+
+            AddStepClickLink("00:00:000 ()", "invalid link");
+            AddAssert("recieved 'failed to process'", () =>
+                Game.Notifications.AllNotifications.Count(x => x.Text == EditorStrings.FailedToProcessTimestamp) == 3
+            );
+
+            AddStepClickLink("00:00:000 (-1)", "invalid link");
+            AddAssert("recieved 'failed to process'", () =>
+                Game.Notifications.AllNotifications.Count(x => x.Text == EditorStrings.FailedToProcessTimestamp) == 4
+            );
+
+            AddStepClickLink("50000:00:000", "too long link");
+            AddAssert("recieved 'too long'", () =>
+                Game.Notifications.AllNotifications.Count(x => x.Text == EditorStrings.TooLongTimestamp) == 1
+            );
+        }
+
+        [Test]
+        public void TestHandleCurrentScreenChanges()
+        {
+            const long long_link_value = 1_000 * 60 * 1_000;
+            RulesetInfo rulesetInfo = new OsuRuleset().RulesetInfo;
+
+            SetUpEditor(rulesetInfo);
+            AddAssert("is editor Osu", () => EditorBeatmap.BeatmapInfo.Ruleset.Equals(rulesetInfo));
+
+            AddStepClickLink("1000:00:000", "long link");
+            AddAssert("moved to end of track", () =>
+                EditorClock.CurrentTime == long_link_value
+                || (EditorClock.TrackLength < long_link_value && EditorClock.CurrentTime == EditorClock.TrackLength)
+            );
+
+            AddStepScreenModeTo(EditorScreenMode.SongSetup);
+            AddStepClickLink("00:00:000");
+            AssertOnScreenAt(EditorScreenMode.SongSetup, 0);
+
+            AddStepClickLink("00:05:000 (0|0)");
+            AssertMovedScreenTo(EditorScreenMode.Compose);
+
+            AddStepScreenModeTo(EditorScreenMode.Design);
+            AddStepClickLink("00:10:000");
+            AssertOnScreenAt(EditorScreenMode.Design, 10_000);
+
+            AddStepClickLink("00:15:000 (1)");
+            AssertMovedScreenTo(EditorScreenMode.Compose);
+
+            AddStepScreenModeTo(EditorScreenMode.Timing);
+            AddStepClickLink("00:20:000");
+            AssertOnScreenAt(EditorScreenMode.Timing, 20_000);
+
+            AddStepClickLink("00:25:000 (0,1)");
+            AssertMovedScreenTo(EditorScreenMode.Compose);
+
+            AddStepScreenModeTo(EditorScreenMode.Verify);
+            AddStepClickLink("00:30:000");
+            AssertOnScreenAt(EditorScreenMode.Verify, 30_000);
+
+            AddStepClickLink("00:35:000 (0,1)");
+            AssertMovedScreenTo(EditorScreenMode.Compose);
+
+            AddStepClickLink("00:00:000");
+            AssertOnScreenAt(EditorScreenMode.Compose, 0);
+        }
+
+        [Test]
+        public void TestSelectionForOsu()
```

Comment:
In line with the other comment, these selection tests should probably go into the ruleset test projects.

Reply:
Separated Test cases o7